src/sentry/api/endpoints/project_commits.py

```diff
@@ -22,12 +22,18 @@ def get(self, request: Request, project) -> Response:
                                           commit belongs to.
         :pparam string project_slug: the slug of the project to list the
                                      commits of.
+        :qparam string query: this parameter can be used to create a
+                              "starts with" filter for the commit key.
         """
+        query = request.GET.get("query")
 
         queryset = Commit.objects.filter(
             organization_id=project.organization_id,
             releasecommit__release__releaseproject__project_id=project.id,
-        ).distinct("id")
+        )
+
+        if query:
+            queryset = queryset.filter(key__istartswith=query)
 
         return self.paginate(
```

Comment:
Would it make sense to order_by ("key", "-date_added").

Is the `query` queryparam optional in this API? If it is, consider doing an order_by like this:

`order_by=("key", "-date_added") if query else ("-date_added")`